internal/controllers/rest_controller.go

```diff
@@ -172,11 +183,99 @@ func (rc RESTController) Logout(w http.ResponseWriter, r *http.Request) {
 	_, err := rc.getCredentials(r)
 	// if a token is not found, return Forbidden.
 	if authErrors.IsTokenNotFound(err) {
-		rc.forbiddenResponse(w, err)
+		rc.forbidden(w, err)
+		return
+	}
+
+	rc.deleteSessionCookie(w, rc.configuration.AccessTokenCookie)
+	w.WriteHeader(http.StatusOK)
+}
+
+// GetToken returns data from the Vault.
+func (rc RESTController) GetToken(c *gin.Context) {
+	r, w := c.Request, c.Writer
+	resource := strings.Split(r.URL.Path, "/")[2]
+	uuid := c.Params.ByName("uuid")
+	path := fmt.Sprintf("%s/data/%s", resource, uuid)
+
+	data, err := rc.provisionTokenService.Get(path)
```

Comment:
this is logic for the service, you don't need to split and so on.

you would have a `GetProfileToken`, or in the `Get`, have a type, for example (prefer the first)

Reply:
Let's use path parameters and define clear paths for each resource/method.

```
GET /profiles/profile-id/tokens
POST /profiles/profile-id/tokens
GET /profiles/profile-id/tokens/token-id
DELETE /profiles/profile-id/tokens/token-id

GET /synthesizers/synthesizer-id/tokens
POST /synthesizers/synthesizer-id/tokens
GET /synthesizers/synthesizer-id/tokens/token-id
DELETE /synthesizers/synthesizer-id/tokens/token-id
```